src/qtoolkit/io/base.py

```diff
@@ -82,9 +83,22 @@ def generate_header(self, options: dict | QResources | None) -> str:
 
         # check that all the options are present in the template
         keys = set(options.keys())
-        extra = keys.difference(template.get_identifiers())
+        all_identifiers = template.get_identifiers()
+        extra = keys.difference(all_identifiers)
         if extra:
-            msg = f"The following keys are not present in the template: {', '.join(sorted(extra))}"
+            close_matches = {}
+            for extra_val in extra:
+                m = difflib.get_close_matches(extra_val, all_identifiers, n=1)
+                if m:
+                    close_matches[extra_val] = m[0]
+            msg = (
+                f"The following keys are not present in the template: {', '.join(sorted(extra))}. "
+                f"Check the template in {type(self).__module__}.{type(self).__qualname__}.header_template."
+            )
+            if close_matches:
+                msg += " Possible replacements:"
+                for extra_val in sorted(close_matches):
+                    msg += f" {close_matches[extra_val]} instead of {extra_val}."
```

Comment:
Maybe the suggested key and the key provided by the user could be between quotes ?

Reply:
done